chia/pools/pool_wallet.py

```diff
@@ -770,12 +785,34 @@ async def claim_pool_rewards(self, fee: uint64) -> TransactionRecord:
             self.log.info(
                 f"Farmer coin: {coin_record.coin} {coin_record.coin.name()} {coin_to_height_farmed[coin_record.coin]}"
             )
-        if len(all_spends) == 0:
+        if len(all_spends) == 0 or the_coin_record_in_question is None:
             raise ValueError("Nothing to claim, no unspent coinbase rewards")
 
-        # No signatures are required to absorb
         spend_bundle: SpendBundle = SpendBundle(all_spends, G2Element())
 
+        # If fee is 0, no signatures are required to absorb
+        full_spend: SpendBundle = spend_bundle
+
+        if fee > 0:
+            get_max_send_amount = await self.standard_wallet.get_max_send_amount()
+            assert get_max_send_amount > fee
+            amount = uint64(0)
+            my_address = await self.standard_wallet.get_new_puzzlehash()
+            fee_spend = await self.standard_wallet.generate_signed_transaction(
+                amount,
+                my_address,
+                fee,
+                None,
+                None,
+                None,
+                False,
+                {Announcement(the_coin_record_in_question.coin.name(), b"$").name()},
```

Comment:
is there an implicit `RESERVE_FEE` condition in here? we need that, right?

Reply:
It's optional, but probably a good idea (although I suppose you could argue it increases the cost slightly).

Reply:
I think the wallet as part of `generate_signed_transaction` will add the RESERVE_FEE condition